### PR TITLE
[MM-28415] - Add spec for public/archive channels search functionality

### PR DESCRIPTION
**Pull request by nevyangelova**
#### Summary
This PR adds an e2e test verifying that users are able to switch between public and archived channels within the 'More channels' modal and are able to search. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28415

### CODE REVIEW

File: e2e/cypress/integration/channel/more_channels_spec.js [hunk 1]
```diff
@@ -141,6 +141,48 @@ describe('Channels', () => {
...
+
+        cy.get('#moreChannelsModal').should('be.visible').within(() => {
+            // * Dropdown should be visible, defaulting to "Public Channels"
+            cy.get('#channelsMoreDropdown').should('be.visible').and('contain', 'Show: Public Channels').click().within((el) => {
+                // # Click on archived channels item
+                cy.findByText('Archived Channels').should('be.visible').click();
+
+                // * Modal should show the archived channels list
+                cy.wrap(el).should('contain', 'Show: Archived Channels');
+            });
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel/more_channels_spec.js:166]: Does not match with the test case. I believe it should stay as "Show: Public Channels".

File: e2e/cypress/integration/channel/more_channels_spec.js [hunk 2]
```diff
@@ -141,6 +141,48 @@ describe('Channels', () => {
...
+
+            // * Users should be able to search
+
+            cy.get('#searchChannelsTextbox').should('be.visible').type('a').wait(TIMEOUTS.HALF_SEC);
+            cy.get('#moreChannelsList').should('be.visible');
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel/more_channels_spec.js:171]: On step 1, it expects "matching Public channels are shown."

File: e2e/cypress/integration/channel/more_channels_spec.js [hunk 3]
```diff
@@ -141,6 +141,48 @@ describe('Channels', () => {
...
+            cy.get('#searchChannelsTextbox').clear();
+
+            // * Users should be able to switch back to public channels
+            cy.get('#channelsMoreDropdown').should('be.visible').click().within((el) => {
+                cy.findByText('Public Channels').should('be.visible').click();
+                cy.wrap(el).should('contain', 'Show: Public Channels');
+            });
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel/more_channels_spec.js:178]: Step 2 is for "Show: Archived Channels".

File: e2e/cypress/integration/channel/more_channels_spec.js [hunk 4]
```diff
@@ -141,6 +141,48 @@ describe('Channels', () => {
...
+
+            // * Users should be able to type and search
+            cy.get('#searchChannelsTextbox').should('be.visible').type('a').wait(TIMEOUTS.HALF_SEC);
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel/more_channels_spec.js:181]: Step 2 does not mention typing any input.

File: e2e/cypress/integration/channel/more_channels_spec.js [hunk 5]
```diff
@@ -141,6 +141,48 @@ describe('Channels', () => {
...
+            cy.get('#moreChannelsList').should('be.visible');
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel/more_channels_spec.js:182]: Should validate "matching Public and Private archived channels are shown."

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @nevyangelova. Please see comments.

File: e2e/cypress/integration/channel/more_channels_spec.js [hunk 6]
```diff
@@ -141,6 +141,42 @@ describe('Channels', () => {
         // * Assert that archived channel doesn't show up in LHS list
         cy.get('#publicChannelList').should('not.contain', testChannel.display_name);
     });
+
+    it('MM-T1702 Search works when changing public/archived options in the dropdown', () => {
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel/more_channels_spec.js:145]: For ease of test maintenance, each test block should be able to run in isolation. Please try to run this with `it.only` and ensure that it still work as expected.

File: e2e/cypress/integration/channel/more_channels_spec.js [hunk 7]
```diff
@@ -141,6 +141,42 @@ describe('Channels', () => {
...
+        cy.apiAdminLogin();
+        cy.apiUpdateConfig({
+            TeamSettings: {
+                ExperimentalViewArchivedChannels: true,
+            },
+        });
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel/more_channels_spec.js:151]: Please add test data "... some public and private archived channels exist" per test case.

File: e2e/cypress/integration/channel/more_channels_spec.js [hunk 8]
```diff
@@ -141,6 +141,42 @@ describe('Channels', () => {
...
+
+        // * Dropdown should be visible, defaulting to "Public Channels"
+        cy.get('#channelsMoreDropdown').should('be.visible').click().within((el) => {
+            cy.findByText('Public Channels').should('be.visible').click();
+            cy.wrap(el).should('contain', 'Show: Public Channels');
+        });
+
+        // * Users should be able to type and search
+        cy.get('#searchChannelsTextbox').should('be.visible').type('a').wait(TIMEOUTS.HALF_SEC);
+        cy.get('#moreChannelsList').should('be.visible');
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel/more_channels_spec.js:166]: Please add verification for "On step #1, matching Public channels are shown."

File: e2e/cypress/integration/channel/more_channels_spec.js [hunk 9]
```diff
@@ -141,6 +141,42 @@ describe('Channels', () => {
...
+        cy.get('#searchChannelsTextbox').clear();
+
+        cy.get('#moreChannelsModal').should('be.visible').within(() => {
+            // * Users should be able to switch to "Archived Channels" list
+            cy.get('#channelsMoreDropdown').should('be.visible').and('contain', 'Show: Public Channels').click().within((el) => {
+                // # Click on archived channels item
+                cy.findByText('Archived Channels').should('be.visible').click();
+
+                // * Modal should show the archived channels list
+                cy.wrap(el).should('contain', 'Show: Archived Channels');
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel/more_channels_spec.js:176]: Please add verification for "On step #2, matching Public and Private archived channels are shown."

**Review by saturninoabril — CHANGES_REQUESTED** [review]
@nevyangelova Please see comments.

**Comment by nevyangelova** [comment]
Sorry @saturninoabril ! I think I finally understood the proper steps! :)

**Review by saturninoabril — APPROVED** [review]
@nevyangelova Thanks for updating and no worries. LGTM, tested and passed.

**Review by sudheerDev — APPROVED** [review]

**Comment by nevyangelova** [comment]
@jgilliam17 reminder to review this

**Review by jgilliam17 — APPROVED** [review]
Thank you @nevyangelova 
Looks good to merge.
- Linked test case details, steps and links updated.

**Pull request merged**